common/service/config/metrics.go

```diff
@@ -86,6 +93,33 @@ func (c *Metrics) NewScope(logger log.Logger) tally.Scope {
 	return tally.NoopScope
 }
 
+func (c *Metrics) newCustomReporterScope(logger log.Logger, customReporter tally.BaseStatsReporter) tally.Scope {
+	reporter, isStatsReporter := customReporter.(tally.StatsReporter)
+	if isStatsReporter {
+		scope, _ := tally.NewRootScope(
+			tally.ScopeOptions{
+				Tags:     c.Tags,
+				Reporter: reporter,
+				Prefix:   c.Prefix,
+			},
+			time.Second)
+		return scope
+	}
+	cachedReporter, isCachedStatsReporter := customReporter.(tally.CachedStatsReporter)
+	if isCachedStatsReporter {
+		scope, _ := tally.NewRootScope(
+			tally.ScopeOptions{
+				Tags:           c.Tags,
+				CachedReporter: cachedReporter,
+				Prefix:         c.Prefix,
+			},
+			time.Second)
+		return scope
+	}
+	logger.Error(fmt.Sprint("Unsupported metrics reporter type: ", reflect.TypeOf(reporter)))
```

Comment:
You should never format strings for the logger. The main point of all those logger is format on demand. Use tags. And btw instead of `reflect.TypeOf(reporter)` you can just use `%T` formatter.